### PR TITLE
Refactor documentation using 11ty

### LINKED ISSUE #294: Remove bookmarklets
After merging `webextension`, the bookmarklet thing will be useless.

To totally remove it:
* [x] remove the form and scripts on documentation website;
* [x] ~~remove cumulative levels files: webextension will allow us to add or remove levels one by one;~~
* [x] ~~remove `set-minimum-level` thing from the Sass layer.~~

This is meant to simplify the Sass part, and reduce the number of files being generated on Sass compilation.

### LINKED ISSUE #359: [website] — improve accessibility
At least external links triggering a new tab should mention it.

And see if there're any other possible improvements.

### LINKED ISSUE #369: Mention JetBrains' support somewhere
In the README.Md and / or on the website itself?

- [ ] [JetBrains' logo](https://www.jetbrains.com/company/brand/logos/)
- [ ] With `https://www.jetbrains.com/?from=a11y.css` as target.

### LINKED ISSUE #391: Use Autoprefixer
Using Autoprefixer in a11y.css would allow to:
1. decrease source file size,
2. let users getting a11y.css through npm or yarn to roll their own Browserlist.

### LINKED ISSUE #395: Docs theming
Right after #376, a few improvements (to-do list moved to #376).

### LINKED ISSUE #403: Switch to GitHub Actions
- [ ] Add Bundlewatch
- [x] Check for compilation (thus drop Travis)
- [x] Ensure doc compiles
- [ ] accessibility testing for docs (see [chaarts #56](https://github.com/ffoodd/chaarts/issues/56))

### LINKED ISSUE #410: Switch to stylelint for both CSS and scss lints
[scss-lint is close to deprecation](https://github.com/sds/scss-lint) and I'm more used to stylelint for years now.

### PR DESCRIPTION
**Pull request by GaetanBt**
Since Hologram has shown its limits and does not seem to be really maintained, we are considering switching to [Eleventy](https://www.11ty.dev/) to build the static documentation of the project.

Fixes #294, fixes #369, fixes #359 

## Get started

`npm ci`, then `npm start` and go.

## To do

- [x] Update Node version to at least v12.10.0 in the Travis configuration
- **Theming**:
  - [x] Revise themes markup (remove duplicated `<h1>`'s, use a list, use `h3` for description, references, etc.)
  - [x] Clean input CSS (some prefixes or IE fixes may not be still useful)
  - [x] improve homepage theming
  - [x] responsive!
- **Content**:
  - [x] run PrismJS on code blocks
    - [x] unescape markup in examples (`&lt;` and `&gt;` everywhere -_-)
    - [x] customize Prism theme,
    - [x] render HTML code blocks too!
  - [x] Drop SassDoc to use 11ty, too:
    - [x] drop Sassdoc comments formalism to use the same markdown-based comments that we did for Hologram
    - [x] hook this up in a11yCSS plugin (`/util/*` as input, `_data/sass/api.json` as output)
    - [x] add PrismJS scss plugin
    - [x] group categories in TOC and content: maybe reconsider the way we generate json source for API content itself?
    - [x] handle multiple matches in ScssContent regex to ensure separate outputs
    - [x] update table / content styles
    - ~~link to GitHub sources for SCSS code blocks?~~ SassDoc 's very own parser kept track of context for each item, which I don't think we're able to (at least automatically).
    - ~~document `Output` / `Return` when appropriate?~~ Or simply drop `Return`?
    - ~~Ensure to document `Requires`~~ Maybe another day :D
    - [x] Link things together (Used by, Requires, Tests, etc.)
    - ~~Private access: should it be displayed?~~
      - maybe another table with return/output and require?
  - [x] Find a way to provide a dynamic `<title>` to data-generated pages
  - [x] Use a Nunjucks component/macro to gather all "external link icons" in one place (→ finally we simply import a SVG file)
  - [x] Revise way to generate docs' CSS which seems to be a little buggy (maybe have a look at [pack11ty](https://pack11ty.dev/documentation/assets/))
- **Cleanup**:
  - [x] Remove all occurrences of "hologram" in code, comments, ...
  - [x] Do not rely on Gulp anymore, thus fixes #410 and fixes #391
  - [x] maybe for CSS ~~& web extension~~ too?
  - [x] Use & update npm scripts and GitHub actions to run, deploy, etc. to fix #403
    - [x] Make current actions pass
    - ~~Run Pa11y-ci or axe-linter?~~ The whole website is meant to have issues, so…
    - [x] Add Bundlewatch
    - [x] Deploy previews using Netlify, as done in Chaarts?
  - [x] build site in `/docs`,
  - [x] set `site` as 11ty folder instead of `src` to prevent confusion
  - [x] Cleanup HTML: duplicate IDs, maybe minimisation too?
  - [x] Improve theming a bit to fix #395 
    - [x] use Autoprefixer for docs CSS
    - [x] switch layout to CSS Grid
    - [x] ~~organize CSS: use Dart Sass, ITCSS, maybe Cube or something?~~
    - [x] handle CSS live reload, after PostCSS + Autoprefixer + CSSnano.
    - [x] RWD (!)
    - [x] ~~use Fira Sans + Fira Code as variable fonts~~ → **The current .woff2 are way lighter thant the available variable fonts**.
    - [x] Refactor CSS completely by using sseeeedd's tricks as CSS basis (eg. custom props for font-stacks and color palettes)
    - [x] Prism theming based on sseeeedd
    - [x] Improve focus handling
    - [x] ~~Remove unused CSS (using PurgeCSS or something?)~~ → **only got a few `.token`s from Prism**.
  - [x] Redirect old `.html` URL's to match the new ones without extension by using the good ol'`refresh` meta, as done [in Bootstrap docs' repo](https://github.com/twbs/bootstrap/blob/gh-pages/docs/index.html). Is that only needed? Maybe a trailing `/` would suffice, see [Amelia Bellamy-Royds explanations recently](https://twitter.com/AmeliasBrain/status/1485068959887998977).
  - [x] Try to upgrade to eleventy 1.0.0 using [eleventy upgrade helper](https://github.com/11ty/eleventy-upgrade-help)?
  - [x] Check [Andy Bell's 11ty-base project](https://github.com/hankchizljaw/11ty-base) for potential improvements.
- [x] Document in README!
- [ ] Rebase and merge :heart:

## After merging

- [ ] deploy `/docs` (in repository settings),
- [ ] then drop `gh-pages` branch and scripts.
- [x] Document in wiki.
- [ ] Drop Travis settings in Webhooks.

## Remaining issues

Not required to fix right away, but would be good to fix at some point. Maybe simply create issues?

- Improve the way to provide a content block without a Sass comment (see `sass/themes/webextension.scss`)
- Document Requires
- Display private access
- Document Returns—and improve Outputs?
- "View source" link

### CODE REVIEW

**Comment by ffoodd** [comment]
For SassDoc:
- probably add a new method based on `parseSassComments()`
- using a new regex based on `commentBlockRegex`.

Then in the method, we'll need several regexes to parse SassDoc annotations.

**Comment by ffoodd** [comment]
Regarding SassDoc, rewriting our whole API Doc using markdown (as for the themes) is more straightforward. Need to define what to for docs with private access: maybe only a flag in their frontmatter, to display info in the output?

**Comment by ffoodd** [comment]
Simple sidenote: [Showdown's wiki is full of useful options doc, and may allow some new tricks](https://github.com/showdownjs/showdown/wiki/Showdown-Options).

**Comment by ffoodd** [comment]
There're a few `&amp;` in Sass / CSS code blocks, need to dig it a bit.

**Comment by ffoodd** [comment]
Translate task does not require json files anymore, which in turn allowed me to simply drop Gulp :tada:

**Comment by ffoodd** [comment]
Maybe missing a few scripts, eg. a build task (sass, autoprefixer, etc.)?

And what about using npm scripts hooks (`pre*` or `post*`)?

**Comment by ffoodd** [comment]
Actions on their way: would  Bundlewatch be useful for this project? I don't think so, any opinion is  welcome!

**Comment by ffoodd** [comment]
Serving 11ty works fine, but the output in `/docs` seems empty… Need to check things out!

File: .github/workflows/webext.yml [hunk 1]
```diff
@@ -18,6 +18,9 @@ jobs:
       - name: Clone repository
         uses: actions/checkout@v2
 
+      - name: Install rename
+        run: sudo apt install rename
```
Comment (wojsmol) [inline .github/workflows/webext.yml:22]: sudo apt update is needed before that line .

Comment (ffoodd) [reply .github/workflows/webext.yml:25]: Indeed, thanks!

**Comment by ffoodd** [comment]
We have duplicate `id`s in HTML output, needs to fix this!

**Comment by ffoodd** [comment]
Deploy preview with Netlify, as done in Chaarts?

**Comment by ffoodd** [comment]
Maybe try to upgrade to eleventy 1.0.0 using [eleventy upgrade helper](https://github.com/11ty/eleventy-upgrade-help)?

**Comment by ffoodd** [comment]
Using [11ty permalinks](https://www.11ty.dev/docs/permalinks/) allowed me to easily ensure that URIs won't change—[which is cool.](https://www.w3.org/Provider/Style/URI.html).

**Pull request merged**